internal/compilation/makefile.py

```diff
@@ -0,0 +1,142 @@
+import subprocess
+import shutil
+import os
+
+from internal.context import TMTContext
+from internal.outcome import (
+    CompilationOutcome,
+    CompilationResult,
+    SingleCompilationResult,
+)
+from internal.runner import Process, wait_for_outputs
+
+from .languages import languages
+
+
+# if platform.system() == "Darwin":
```

Comment:
Shall we ask user to ensure their make is `gmake`, or shall we auto-detect and choose here?